plugins/module_utils/network/sonic/config/route_maps/route_maps.py

```diff
@@ -1424,12 +1476,21 @@ def get_route_map_delete_set_bgp(self, command, set_both_keys, cmd_rmap_have, re
                     if cfg_set_top['extcommunity'].get(extcomm_type):
                         # Append eligible entries to the delete list. Remember which entries
                         # are ineligible.
-                        for extcomm_number in cmd_set_top['extcommunity'][extcomm_type]:
-                            if extcomm_number in cfg_set_top['extcommunity'][extcomm_type]:
-                                set_extcommunity_delete_attrs.append(
-                                    self.set_extcomm_rest_names[extcomm_type] + extcomm_number)
+                        if extcomm_type == "bandwidth":
+                            bandwidth_value = cfg_set_top['extcommunity'][extcomm_type]["bandwidth_value"]
+                            transitive_value = cfg_set_top['extcommunity'][extcomm_type]["transitive_value"]
+                            if transitive_value:
+                                transitive_string = "transitive"
                             else:
-                                ext_comm_number_remove_list.append(extcomm_number)
+                                transitive_string = "non-transitive"
+                            if (bandwidth_value == cmd_set_top['extcommunity'][extcomm_type].get("bandwidth_value") and
+                               transitive_value == cmd_set_top['extcommunity'][extcomm_type].get("transitive_value")):
+                                set_extcommunity_delete_attrs.append(self.set_extcomm_rest_names[extcomm_type] + bandwidth_value + ":" + transitive_string)
```

Comment:
If this 'bandwidth' extcommunity is not eligible for deletion, remove it from the list of deleted extcommunities.


```suggestion
                            if (bandwidth_value == cmd_set_top['extcommunity'][extcomm_type].get("bandwidth_value") and
                               transitive_value == cmd_set_top['extcommunity'][extcomm_type].get("transitive_value")):
                                set_extcommunity_delete_attrs.append(self.set_extcomm_rest_names[extcomm_type] + bandwidth_value + ":" + transitive_string)
                            else:
                                cmd_set_top['extcommunity'].pop('bandwidth')
```

Reply:
Addressed in https://github.com/awhaley-dell/dellemc.enterprise_sonic/commit/01c7fcf52c3ef60e5dae01f853fa6e5f06819ebe.